src/main/java/com/example/demo/controller/F1DriverStatsController.java

```diff
@@ -0,0 +1,144 @@
+import com.example.demo.model.Team;
+import com.example.demo.repository.DriverRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.swing.*;
+import java.util.Optional;
+
+import static org.springframework.http.ResponseEntity.notFound;
+import static org.springframework.http.ResponseEntity.ok;
+
+@RestController
+@RequestMapping("/f1stats")
+@RequiredArgsConstructor
+public class F1DriverStatsController {
+    private final DriverRepository driverRepository;
+
+    @PostMapping(path = "/add")
+    public @ResponseBody
+    ResponseEntity<Driver> addDriver(@RequestParam String firstname,
+                                     @RequestParam String lastname,
+                                     @RequestParam int wins,
+                                     @RequestParam int poles,
+                                     @RequestParam int fastest,
+                                     @RequestParam double points) {
+        Team team = Team.builder()
+                .teamName("Ferrari")
+                .build();
+        Driver driver = Driver.builder().currentTeam(team).build();
+        driver.setFirstName(firstname);
+        driver.setLastName(lastname);
+        driver.setWins(wins);
+        driver.setPoles(poles);
+        driver.setFastestLaps(fastest);
+        driver.setPoints(points);
+        driverRepository.save(driver);
+        return new ResponseEntity<>(driver, HttpStatus.CREATED);
+    }
+
+    @PostMapping(path = "/addDriver")
+    public @ResponseBody
+    ResponseEntity<Driver> addDriver(@RequestBody Driver driver) {
+        Driver savedDriver = driverRepository.save(driver);
+        savedDriver.setCurrentTeam(Team.builder().teamName("Ferrari").build());
+        return new ResponseEntity<>(savedDriver, HttpStatus.CREATED);
+    }
+
+    @GetMapping(path = "/all")
+    public Iterable<Driver> getAllDrivers() {
+        return driverRepository.findAll();
+    }
+
+    @GetMapping(path = "/{id}")
+    public Optional<Driver> getDriverById(@PathVariable Long id) {
+        return driverRepository.findById(id);
+    }
+
+    @GetMapping(path = "/get-all-firstname-{firstName}")
+    public Iterable<Driver> getDriversWithFirstName(@PathVariable String firstName) {
+        return driverRepository.findAllByFirstName(firstName);
+    }
+
+    @GetMapping(path = "/get-all-lastname-{lastName}")
+    public Iterable<Driver> getDriversWithLastName(@PathVariable String lastName) {
+        return driverRepository.findAllByLastName(lastName);
+    }
+
+    @GetMapping(path = "/most-wins")
+    public Iterable<Driver> getDriverWithMostWins() {
+        return driverRepository.findFirstByOrderByWinsDesc();
+    }
+
+    @GetMapping(path = "/wins")
+    public @ResponseBody
+    Iterable<Driver> winsBetween(@RequestParam int from, @RequestParam int to) {
```

Comment:
This again can be find?winsLessThan={}&winsMoreThan{} 

a pseudo search on a few fields is a nice little adjustment to the task. Please combine them